sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/V2SessionCatalog.scala

```diff
@@ -171,8 +171,6 @@ class V2SessionCatalog(catalog: SessionCatalog, conf: SQLConf)
       ident.namespace match {
         case Array(db) =>
           TableIdentifier(ident.name, Some(db))
-        case Array() =>
```

Comment:
This is the actual change that removes filling the db automatically.

Reply:
shall we give a better error message when database is not present?

Reply:
Yes, good idea!